server/plugins/builder/pages/index.js

```diff
@@ -19,39 +19,49 @@ class Page extends EngineBasePage {
     return {
       payload: {
         output: 'stream',
-        parse: true
+        parse: true,
+        maxBytes: 5e+6 // 5mb
       },
       ext: {
         onPreHandler: {
           method: async (request, h) => {
             let files = fileStreamsFromPayload(request.payload)
-            if (!files.length) {
-              return h.continue
-            }
-            let file = files[0]
-            let key = file[0]
-            let previousUpload = (request.yar.get('originalFilenames') || {})[key]
-            if (previousUpload && file[1].hapi.filename === '') {
-              h.request.payload[key] = previousUpload.location
-              return h.continue
-            }
-            try {
-              // TODO:- should be for each(?) or limit to one upload per request..? 🤔
-              let saved = await saveFileToTmp(file[1])
-              let { error, location } = await uploadDocument(saved)
-              if (location) {
-                request.yar.set('originalFilenames', { [key]: { originalFilename: file[1].hapi.filename, location } })
-                h.request.payload[key] = location
+            let state = await this.model.getState(request)
+            let originalFilenames = state.originalFilenames || {}
+
+            files.forEach(file => {
+              let key = file[0]
+              let previousUpload = originalFilenames[key]
+              if (previousUpload && file[1]._data.length) {
```

Comment:
Not sure about this... if we have a previous upload and new data has been submitted, use the old data? Otherwise delete the new data?

Reply:
oop meant to be < 1 check..

Reply:
The delete needs to be wrapped in the <1 check as well?